### PR TITLE
[Enhancement] add session variable default_table_compression

### LINKED ISSUE #17642: Add a setting for default table compression algorithm
Per Slack discussion, could we have a setting default_data_compression_algorithm ?

This would enable us to default to ZSTD, etc.  to reduce footprint instead of consistently having to specify.

### PR DESCRIPTION
**Pull request by murphyatwork**
Signed-off-by: Murphy <mofei@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17642

## Problem Summary(Required) ：

Add a session variable `default_table_compression` to choose default compression algorithm when creating table.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/CompressionUtils.java|0|1|00.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|1|6|16.67%|[180, 181, 182, 183, 184]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|9|9|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by imay — DISMISSED** [review]

File: docs/reference/System_variable.md [hunk 1]
```diff
@@ -104,6 +104,10 @@ SELECT /*+ SET_VAR(query_timeout = 1) */ sleep(3);
 
   Used to set the default storage format used by the storage engine of the computing node. The currently supported storage formats are `alpha` and `beta`.
 
+* default_table_compression
+
+  Set the default compression algorithm for table storage, supported compression algorithms are: `snappy, lz4, lz4_frame, zlib, zstd, gzip, deflate, bzip2, lzo, brotli`.
```
Comment (imay) [inline docs/reference/System_variable.md:109]: AFAIK we don't support deflate, lzo, brotli

Comment (mofeiatwork) [reply docs/reference/System_variable.md:109]: fixed

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/CompressionUtils.java|0|1|00.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|1|6|16.67%|[180, 181, 182, 183, 184]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|9|9|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/CompressionUtils.java|0|1|00.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|1|6|16.67%|[180, 181, 182, 183, 184]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|9|9|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: docs/reference/System_variable.md [hunk 2]
```diff
@@ -104,6 +104,10 @@ SELECT /*+ SET_VAR(query_timeout = 1) */ sleep(3);
 
   Used to set the default storage format used by the storage engine of the computing node. The currently supported storage formats are `alpha` and `beta`.
 
+* default_table_compression
+
+  Set the default compression algorithm for table storage, supported compression algorithms are: `snappy, lz4, lz4_frame, zlib, zstd`.
```
Comment (chaoyli) [inline docs/reference/System_variable.md:109]: LZ4_FRAME is not necessary to provide.

Comment (mofeiatwork) [reply docs/reference/System_variable.md:109]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/CompressionUtils.java|0|1|00.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|1|6|16.67%|[180, 181, 182, 183, 184]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|9|9|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/CompressionUtils.java|0|1|00.00%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SetStmtAnalyzer.java|1|6|16.67%|[180, 181, 182, 183, 184]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|9|9|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Pull request merged**